crates/nu-protocol/src/shell_error.rs

```diff
@@ -1085,6 +1085,72 @@ pub enum ShellError {
         #[label("not a boolean expression")]
         span: Span,
     },
+
+    /// An attempt to run a command marked for constant evaluation lacking the const. eval.
+    /// implementation.
+    ///
+    /// This is an internal Nushell error, please file an issue.
+    #[error("Missing const eval implementation")]
+    #[diagnostic(
+        code(nu::shell::missing_const_eval_implementation),
+        help(
+            "The command lacks an implementation for constant evaluation. \
+This is an internal Nushell error, please file an issue https://github.com/nushell/nushell/issues."
+        )
+    )]
+    MissingConstEvalImpl {
+        #[label("command lacks constant implementation")]
+        span: Span,
+    },
+
+    /// Tried assigning non-constant value to a constant
+    ///
+    /// ## Resolution
+    ///
+    /// Only a subset of expressions are allowed to be assigned as a constant during parsing.
+    #[error("Not a constant.")]
+    #[diagnostic(
+        code(nu::shell::not_a_constant),
+        help("Only a subset of expressions are allowed constants during parsing. Try using the 'const' command or typing the value literally.")
+    )]
+    NotAConstant(#[label = "Value is not a parse-time constant"] Span),
+
+    /// Tried running a command that is not const-compatible
+    ///
+    /// ## Resolution
+    ///
+    /// Only a subset of builtin commands, and custom commands built only from those commands, can
+    /// run at parse time.
+    #[error("Not a const command.")]
+    #[diagnostic(
+        code(nu::shell::not_a_const_command),
+        help("Only a subset of builtin commands, and custom commands built only from those commands, can run at parse time.")
+    )]
+    NotAConstCommand(#[label = "This command cannot run at parse time."] Span),
+
+    /// Tried getting a help message at parse time.
+    ///
+    /// ## Resolution
+    ///
+    /// Help messages are not supported at parse time.
+    #[error("Help message not a constant.")]
+    #[diagnostic(
+        code(nu::shell::not_a_const_help),
+        help("Help messages are currently not supported to be constants.")
+    )]
+    NotAConstHelp(#[label = "Cannot get help message at parse time."] Span),
+}
+
+// TODO: Implement as From trait
```

Comment:
not sure if you can, since you need multiple params

Reply:
What do you mean by needing multiple params? Do you mean that it requires some runtime values?

Reply:
@kubouch - From only takes one param but you use self, working_set, and span below.

https://doc.rust-lang.org/std/convert/trait.From.html

Reply:
I see, now I get what you were commening on :-D. I think `wrap` is good enough. It's a bit of a hack but the error messages actually look quite decent.